.github/workflows/test.yaml

```diff
@@ -492,7 +492,10 @@ jobs:
   # The files are published only when a tag is created.
   release-publish:
     name: Check release and publish on twisted-* tag
-    runs-on: 'ubuntu-22.04'
+    runs-on: ubuntu-latest
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing
+      id-token: write
```

Comment:
write token should not be available to the build/checkout phase

Reply:
I started work here before I saw you adding it https://github.com/twisted/twisted/pull/12379 can copy paste my changes in if you agree with them?

Reply:
Thanks for the PR. I think that your PR is very useful.

I added the publishing changes to this PR, as otherwise, I don't know how we can test the changes.

As part of the relase PR, we can create various release candiates and trigger a real PyPI publish process

---

I have just copied to code that we use for twisted/towncrier publishing.
Trying to keep it simple :)